WalletWasabi.Gui/Tabs/WalletManager/LoadWalletViewModel.cs

```diff
@@ -127,74 +188,136 @@ public override void OnCategorySelected()
 			var walletFiles = directoryInfo.GetFiles("*.json", SearchOption.TopDirectoryOnly).OrderByDescending(t => t.LastAccessTimeUtc);
 			foreach (var file in walletFiles)
 			{
-				_wallets.Add(Path.GetFileNameWithoutExtension(file.FullName));
+				Wallets.Add(Path.GetFileNameWithoutExtension(file.FullName));
 			}
 
-			if (_wallets.Any())
+			SelectedWallet = Wallets.FirstOrDefault();
+			SetWalletStates();
+		}
+
+		private void SetWalletStates()
+		{
+			IsWalletSelected = !string.IsNullOrEmpty(SelectedWallet);
+			CanTestPassword = IsWalletSelected;
+
+			IsWalletOpened = Global.WalletService != null;
+			// If not busy loading.
+			// And wallet is selected.
+			// And no wallet is opened.
+			CanLoadWallet = !IsBusy && IsWalletSelected && !IsWalletOpened;
+
+			if (IsWalletOpened)
 			{
-				SelectedWallet = _wallets.First();
+				SetWarningMessage("There is already an open wallet. Restart the application to open another one.");
 			}
-
-			IsWalletOpened = !(Global.WalletService is null);
-			ValidationMessage = null;
 		}
 
 		public ReactiveCommand LoadCommand { get; }
+		public ReactiveCommand TestPasswordCommand { get; }
 
-		public async Task LoadWalletAsync()
+		public KeyManager LoadKeyManager(bool requirePassword)
 		{
 			try
 			{
-				IsBusy = true;
+				CanTestPassword = false;
+				var password = Guard.Correct(Password); // Don't let whitespaces to the beginning and to the end.
+				Password = ""; // Clear password field.
+
+				if (string.IsNullOrEmpty(SelectedWallet))
+				{
+					SetValidationMessage("No wallet selected.");
+					return null;
+				}
 
 				var walletFullPath = Path.Combine(Global.WalletsDir, SelectedWallet + ".json");
 				var walletBackupFullPath = Path.Combine(Global.WalletBackupsDir, SelectedWallet + ".json");
 				if (!File.Exists(walletFullPath) && !File.Exists(walletBackupFullPath))
 				{
 					// The selected wallet is not available any more (someone deleted it?).
 					OnCategorySelected();
-					SelectedWallet = null;
-					ValidationMessage = "The selected wallet and its backup doesn't exsist, did you delete them?";
-					return;
+					SetValidationMessage("The selected wallet and its backup doesn't exsist, did you delete them?");
```

Comment:
There is a typo here **exsist** -> exist. Also, I am not the best in English but I think it should say **don't** instead of **doesn't** but I am not really sure.

Reply:
Yes, it's plural. Fixed.